### PR TITLE
[bazel] fix libc build

### PR DESCRIPTION
**Pull request by nickdesaulniers**
Link: #118899

### CODE REVIEW

**Review by keith — APPROVED** [review]
BUILD change looks good, i would probably land that first and wait for CI since who knows what else might break with bazel 8.x

File: utils/bazel/llvm-project-overlay/libc/test/src/strings/BUILD.bazel
Comment (nickdesaulniers) [inline utils/bazel/llvm-project-overlay/libc/test/src/strings/BUILD.bazel:33]: This isn't quite right, not sure the correct incantation:

```
ERROR: no such package '@@llvm-project//libc/test/string': BUILD file not found in directory 'libc/test/string' of external repository @@llvm-project. Add a BUILD file to a directory to mark it as a package.
ERROR: /usr/local/google/home/ndesaulniers/.cache/bazel/_bazel_ndesaulniers/02cf1b716d72de9bb7498bab3b083b83/external/llvm-project/libc/test/src/strings/BUILD.bazel:39:10: no such package '@@llvm-project//libc/test/string': BUILD file not found in directory 'libc/test/string' of external repository @@llvm-project. Add a BUILD file to a directory to mark it as a package. and referenced by '@@llvm-project//libc/test/src/strings:bcmp_test'
ERROR: Analysis of target '@@llvm-project//libc/test/src/strings:bcmp_test' failed; build aborted: Analysis failed
```

File: utils/bazel/llvm-project-overlay/libc/test/src/strings/BUILD.bazel
Comment (michaelrj-google) [inline utils/bazel/llvm-project-overlay/libc/test/src/strings/BUILD.bazel:60]: Need to add `/src` between `/test` and `/string`. Also format the dependencies.
```suggestion

libc_test(
    name = "bcopy_test",
    srcs = ["bcopy_test.cpp"],
    libc_function_deps = [
        "//libc:bcopy",
    ],
    deps = [
        "//libc:__support_cpp_span",
        "//libc/test/UnitTest:memory_matcher",
        "//libc/test/src/string:memory_check_utils",
    ],
)

libc_test(
    name = "bcmp_test",
    srcs = ["bcmp_test.cpp"],
    libc_function_deps = [
        "//libc:bcmp",
    ],
    deps = [
        "//libc/test/UnitTest:test_logger",
        "//libc/test/src/string:memory_check_utils",
    ],
)

libc_test(
    name = "bzero_test",
    srcs = ["bzero_test.cpp"],
    libc_function_deps = [
        "//libc:bzero",
    ],
    deps = [
        "//libc/test/src/string:memory_check_utils",
    ],
)
```

File: utils/bazel/llvm-project-overlay/libc/test/src/strings/BUILD.bazel
Comment (nickdesaulniers) [reply utils/bazel/llvm-project-overlay/libc/test/src/strings/BUILD.bazel]: 221a65def345

File: utils/bazel/llvm-project-overlay/libc/test/src/strings/BUILD.bazel
Comment (nickdesaulniers) [reply utils/bazel/llvm-project-overlay/libc/test/src/strings/BUILD.bazel]: 221a65def345 for the paths

**Comment by nickdesaulniers** [comment]
Going for lunch, will merge after. carte blanche to merge sooner otherwise.

**Pull request merged**
